### ISSUE TITLE
dataset and preprocessing for tacotron2 + lpcnet

### ISSUE THREAD

**Issue by alokprasad** [MlWoo/LPCNet#3]
1.Do we need to train LPCNET with LJSpeech dataset or 16k-LP7?
2.DO we need to train both LPCNET and tacatron2 with same dataset?
3.Do we need to Tacatron-2/preprocess.py or just use
./header_removal.sh-->pcm files
./feature_extract.sh-->f32 files
train.py takes wavand npy files how  will it takes pcm and f32 file.?

**Comment by MlWoo** [comment]
0. Every datast has their own data distribution.
2. same dataset is recommended.
1. I think LPCNet with 16k-LP7 will recover waveform from the audio feature which predicted from tacotron2 no matter which dataset is trained. Howerver, it maybe do harm to the effect when your train the two models with different datasets. refer 0st and 2nd point.
3. refer to issure #4 .

extrally. T2 maybe needs a large dataset. So, training them with large dataset such as LJspeech is good idea.

**Comment by lyz04551** [comment]
For tacotron docking lpcnet, directly set the mel spectrum dimension to 20, to dock the 20 features of the lpcnet input, is this not appropriate. From the effect point of view, you can hear the complete sound, but not perfect. I also want to trouble you.

**Comment by lyz04551** [comment]
Retrain Tacotron2+lpcnet and train lpcnet to use GTA mode？

**Comment by MlWoo** [comment]
@lyz04551 I  do not suggest to use GTA mode if you have not read the LPCNet code deeply. You should figure out the start and end of the audio or other transformation introduced by feature extractation of LPCNet.

**Comment by lyz04551** [comment]
> @lyz04551 I do not suggest to use GTA mode if you have not read the LPCNet code deeply. You should figure out the start and end of the audio or other transformation introduced by feature extractation of LPCNet.

Thank you for your reply. Have you encountered any problems with this synthetic audio volume?
I use the LPCTron code of alokprasad. The basic parameters are the same. It is the same as the training method recommended by you. The Chinese data set used is retrained. Do you have a similar volume problem?
![image](https://user-images.githubusercontent.com/27425887/62672028-ad63a880-b9cb-11e9-9627-ed2d70a02c12.png)

**Comment by MlWoo** [comment]
@lyz04551  You could normalize the audio before features extraction.

**Comment by lyz04551** [comment]
> @lyz04551 You could normalize the audio before features extraction.

Can you explain a little more about how to do it, thank you. I am afraid that my approach caused a big mistake.

**Comment by lyz04551** [comment]
> @lyz04551 You could normalize the audio before features extraction.

What can be said in detail, is the normalization of the magnitude?

**Comment by MlWoo** [comment]
@lyz04551  yes, rescale the audio volume which you can refer to the tacotron2 preprocessing.

**Comment by lyz04551** [comment]
> @lyz04551 yes, rescale the audio volume which you can refer to the tacotron2 preprocessing.

Sorry, I am coming again. After your guidance, I feel that the volume of this training is quite normal, but I feel that the audio starts with a little noise. The overall feeling is not as good as the sound quality of the waves. Is my training not fully converged now? I use Tacotron-2 32000step and lpcnet22epoch synthesized speech. I would like to ask you how many steps you can train in the two models to synthesize good sounds without any post-processing? The last question, my lpcnet model loss seems to drop very slowly, from the initial 3.5 to the current 3.2?

**Comment by lyz04551** [comment]
> @lyz04551 yes, rescale the audio volume which you can refer to the tacotron2 preprocessing.

Sorry, I am coming again. After your guidance, I feel that the volume of this training is quite normal, but I feel that the audio starts with a little noise. The overall feeling is not as good as the sound quality of the wavenet. Is my training not fully converged now? I use Tacotron-2 32000step and lpcnet 22 epoch synthesized speech. I would like to ask you how many steps you can train in the two models to synthesize good sounds without any post-processing? The last question, my lpcnet model loss seems to drop very slowly, from the initial 3.5 to the current 3.2?

**Comment by lyz04551** [comment]
> @lyz04551 yes, rescale the audio volume which you can refer to the tacotron2 preprocessing.


[test.zip](https://github.com/MlWoo/LPCNet/files/3511786/test.zip)
This is the sound that I combined with Tacotron+lpcnet. I always feel that the sound quality is not particularly good. Can you hear some questions?

**Comment by carlfm01** [comment]
Hello @lyz04551 did you fixed the issue? I'm facing the same.

Examples here from taco2 and from the lpc extracted, it also contains my hparams :
[audios.zip](https://github.com/MlWoo/LPCNet/files/3588213/audios.zip)

Which params am I missing? Is the same datasets with the same extracted features, any idea?

From the real features looks good.

![Captura de pantalla (918)](https://user-images.githubusercontent.com/32177100/64496541-2bcd9780-d263-11e9-9370-bf2d53b06e8c.png)

From taco2 looks like is applying a filter, but can't figure out where.
![Captura de pantalla (919)](https://user-images.githubusercontent.com/32177100/64496563-73ecba00-d263-11e9-852c-95fa8c46dbbd.png)

Thanks @MlWoo and @alokprasad for the scripts, saved me a lot of time.
